modules/service/src/main/scala/lucuma/odb/service/ObservationService.scala

```diff
@@ -645,6 +709,214 @@ object ObservationService {
+          c_hour_angle_max,
+          c_science_mode,
+          c_spec_wavelength,
+          c_spec_resolution,
+          c_spec_signal_to_noise,
+          c_spec_signal_to_noise_at,
+          c_spec_wavelength_coverage,
+          c_spec_focal_plane,
+          c_spec_focal_plane_angle,
+          c_spec_capability,
+          c_observing_mode_type
+        )
+        SELECT 
+          c_program_id,
+          c_title,
+          c_subtitle,
+          c_instrument,
+          'new',
+          'active',
+          c_visualization_time,
+          c_pts_pi,
+          c_pts_uncharged,
+          c_pts_execution,
+          c_pac_mode,
+          c_pac_angle,
+          c_explicit_ra,
+          c_explicit_dec,
+          c_cloud_extinction,
+          c_image_quality,
+          c_sky_background,
+          c_water_vapor,
+          c_air_mass_min,
+          c_air_mass_max,
+          c_hour_angle_min,
+          c_hour_angle_max,
+          c_science_mode,
+          c_spec_wavelength,
+          c_spec_resolution,
+          c_spec_signal_to_noise,
+          c_spec_signal_to_noise_at,
+          c_spec_wavelength_coverage,
+          c_spec_focal_plane,
+          c_spec_focal_plane_angle,
+          c_spec_capability,
+          c_observing_mode_type
+      FROM t_observation
+      WHERE c_observation_id = $observation_id
+      """.apply(oid) |+|
+      ProgramService.Statements.existsUserAccess(user, pid).foldMap(void"AND " |+| _) |+|
+      void"""
+        RETURNING c_observation_id
+      """
+
+    def cloneAsterism(oldOid: Observation.Id, newOid: Observation.Id): AppliedFragment =
+      sql"""
+        INSERT INTO t_asterism_target (
+          c_program_id,
+          c_observation_id,
+          c_target_id
+        )
+        SELECT 
+          t_asterism_target.c_program_id,
+          $observation_id,
+          t_asterism_target.c_target_id
+        FROM t_asterism_target
+        JOIN t_target ON t_target.c_target_id = t_asterism_target.c_target_id
+        WHERE c_observation_id = $observation_id
+        AND t_target.c_existence = 'present' -- don't clone references to deleted targets
+      """.apply(newOid ~ oldOid)
+    
+    def cloneGmosNorthStatic(oldOid: Observation.Id, newOid: Observation.Id): AppliedFragment =
+      sql"""
+        INSERT INTO t_gmos_north_static (
+          c_observation_id,
+          c_instrument,
+          c_detector,
+          c_mos_pre_imaging,
+          c_nod_and_shuffle,
+          c_stage_mode
+        )
+        SELECT 
+          $observation_id,
+          c_instrument,
+          c_detector,
+          c_mos_pre_imaging,
+          c_nod_and_shuffle,
+          c_stage_mode
+        FROM t_gmos_north_static
+        WHERE c_observation_id = $observation_id
+      """.apply(newOid ~ oldOid)
+
+    def cloneGmosNorthDynamic(oldOid: Observation.Id, newOid: Observation.Id): AppliedFragment =
+      sql"""
+        INSERT INTO t_gmos_north_dynamic (
```

Comment:
There is an `ObservingModeServices`, which I'd thought could be used to manipulate the specific table for the observing mode via the matching service.  For example, `GmosLongSlitService` for now with others added as new modes are introduced.  It's used elsewhere in `ObservationService` like: https://github.com/gemini-hlsw/lucuma-odb/blob/main/modules/service/src/main/scala/lucuma/odb/service/ObservationService.scala#L213-L256.

Reply:
Ok thanks, I didn't know about this. :+1: